### DISCUSSION TITLE
Saving as non incremental PDF, with the same name

### DISCUSSION THREAD

**Discussion by ovidioruzzier** [pymupdf/PyMuPDF#1851] [Q&A] [answered]
I open a PDF file
```
>>> doc = fitz.open('myPreciousPDF.pdf')
```
I create a new doc
```
>>> doc2 = fitz.open()
```
then I insert/add pages to doc2 from doc.
It's a new object and I set its name to a new one,
```
>>> doc2.name = 'myPreciousPDF_modified.pdf'
```
with a reference in its filename to the original PDF file, so I know from which file its content comes from.
Hence
```
>>> doc2.name
'myPreciousPDF_modified.pdf'
```
Then if I do
```
>>> doc2.save(doc.name)
```
it complains because it asks for incremental save, but if I do
```
>>> doc2.save()
```
or
```
>>> doc2.saveIncr()
```
or
```
>>> doc2.save(doc2.name, incremental=True)
```
it complains because it's a new doc.
I would like the name, to be used for the file on disk, to go along with the new doc and save it without the need to store the final filename in another variable.

How to work around this issue?

TIA,

Ovidio

**Comment by JorjMcKie** [comment]
> It's a new object and I set its name to a new one,
> 
> >>> doc2.name = 'myPreciousPDF_modified.pdf'

... and this is the problem! Leave the name of the new document **_empty_** - instead **_save_** the new file with the desired new name.

**Reply by ovidioruzzier** [reply]
Thanks @JorjMcKie ,

the idea behind, maybe I didn't express it correctly, is that I have 5-6 modules each performing a different action on a PDF, like, rotating, graying all the pictures, adding a logo and depending on the source document only some actions are performed, I wanted to reflect the status of the modifications by using a suffix in the filename. That's why I thought as the simplest idea to modify the attribute doc.name.
From your reply it seems it's not a viable way.
Is the only option I have,to create a class derived from fitz.doc  (I hope to use the right terms) and add an attribute that keeps track of this path of modification and override the save method to use the suffix built along the modification done on the PDF document?
I'm asking just to see if I can achieve my goal just by using what fits provides; if not I would bo what I have just described above.

Thanks,

Ovidio

**Comment by JorjMcKie** [comment, marked as answer]
I understand your goal, and it is achievable in the way I described:
1. create a new document
2. make addition of pages, changes of all sorts etc. - also by passing arounbd the document object to other modules.
3. Once you are done and want to save that new document, do `doc2.save("myPreciousPDF_modified.pdf", <any options>)`. **_Do not ever touch doc2.name!_**
